### PR TITLE
Fix false positive: skip screen-reader-only elements in text_size_too_small check

### LINKED ISSUE #1688: Elementor icon screen reader text (or something) false flagging text too small
Image
Image
Image

This is on the social icons on Xeo - I'm unsure why icons like this would flag text as too small. Can the rule be improved at all?

### PR DESCRIPTION
**Pull request by SteveJonesDev**
## Summary

- Adds `isScreenReaderOnly()` helper to `helpers.js` that detects the sr-only CSS technique (classic `clip: rect(0px, 0px, 0px, 0px)` and modern `clip-path: inset(50%)`) by walking the element and its ancestors
- Updates `text-size-too-small.js` to bail out early when the node is screen-reader-only, preventing false positives on visually-hidden accessible labels
- Fixes a false positive triggered by Elementor's social icon widget, where `<span class="elementor-screen-only">` labels were being flagged for small font size

## Context

Elementor (and other builders) use a visually-hidden pattern to provide accessible text labels for icon-only elements. These spans inherit a small computed font size from the icon container but are never visible to sighted users — enforcing a minimum font size on them is meaningless and incorrect.

Axe-core has internal `_isVisibleOnScreen` / `_isVisibleToScreenReaders` functions that could theoretically address this, but they are undocumented internal APIs not exposed via `axe.commons`, operate on virtual nodes rather than DOM nodes, and could change without notice. A thin custom helper over `getComputedStyle` is the safer approach.

## Test plan

- [ ] Verify an Elementor social icons widget no longer triggers Text Too Small
- [ ] Verify genuinely small visible text still triggers the check
- [ ] Verify `display: none` / `visibility: hidden` elements are unaffected (handled upstream by axe)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Text-size validation now excludes screen-reader-only (visually hidden) elements from font-size checks, preventing false positives for hidden content.

* **New Features**
  * Added helper to reliably detect screen-reader-only clipping patterns so accessibility checks focus on visible text.

* **Tests**
  * Added comprehensive tests validating the text-size accessibility rule across visible, hidden, and edge-case scenarios.

Fixes: #1688

[Review Change Stack](https://app.coderabbit.ai/change-stack/equalizedigital/accessibility-checker/pull/1690)

### CODE REVIEW

**Comment by SteveJonesDev** [comment]
Addressed the CodeRabbit nitpick — `isScreenReaderOnly` is now deferred into the leaf/text-node branch so ancestor style walks only happen when we're actually about to evaluate font size.

Skipping the two Gemini suggestions:
- **Position verification on `clip`** — adding a position check would tighten the pattern but also miss non-standard sr-only implementations that use clip without absolute/fixed positioning. The clip value alone is a reliable enough signal for this use case.
- **Shadow DOM traversal** — `parentElement` already stops at shadow boundaries, so the current helper won't walk out of a shadow tree. Supporting sr-only ancestors inside a shadow host is a separate concern and out of scope here.

File: src/pageScanner/helpers/helpers.js [hunk 1]
```diff
@@ -52,6 +52,26 @@ export const isElementVisible = ( element ) => {
...
+ * Font-size checks are irrelevant for such elements.
+ *
+ * @param {HTMLElement} element The element to check.
+ * @return {boolean} True if the element is screen-reader-only.
+ */
+export const isScreenReaderOnly = ( element ) => {
+	let el = element;
+	while ( el && el.nodeType === Node.ELEMENT_NODE ) {
+		const style = window.getComputedStyle( el );
+		if ( style.clip === 'rect(0px, 0px, 0px, 0px)' || style.clipPath === 'inset(50%)' ) {
```
[the remark this answers is not in the document]
Comment (SteveJonesDev) [reply src/pageScanner/helpers/helpers.js]: Good catch — applied. `clip` is only effective on `position: absolute/fixed` elements, so the position guard makes this more correct. `clipPath: inset(50%)` is left unguarded since it's not subject to the same constraint.

File: src/pageScanner/helpers/helpers.js [hunk 2]
```diff
@@ -52,6 +52,26 @@ export const isElementVisible = ( element ) => {
...
+			return true;
+		}
+		el = el.parentElement;
+	}
```
[the remark this answers is not in the document]
Comment (SteveJonesDev) [reply src/pageScanner/helpers/helpers.js:78]: Valid limitation — `parentElement` does stop at shadow boundaries. Shadow DOM support is out of scope for this fix but noted; added to issue #1691 which tracks broader hidden-element detection cleanup.

**Review by pattonwebz — APPROVED** [review]
Tested this on playground and confirmed it solves the issue when font size is `0`.
